src/Tasks/Microsoft.NET.Build.Tasks/DependencyContextBuilder.cs

```diff
@@ -562,6 +579,26 @@ private IEnumerable<CompilationLibrary> GetDirectReferenceCompilationLibraries()
                 Enumerable.Empty<CompilationLibrary>();
         }
 
+        private IEnumerable<RuntimeLibrary> GetDirectReferenceRuntimeLibraries()
+        {
+            return GetReferenceRuntimeLibraries(_directReferences, "reference");
+        }
+
+        private IEnumerable<CompilationLibrary> GetDirectReferenceCompilationLibraries()
+        {
+            return GetReferenceCompilationLibraries(_directReferences, "reference");
+        }
+
+        private IEnumerable<RuntimeLibrary> GetDependencyReferenceRuntimeLibraries()
+        {
+            return GetReferenceRuntimeLibraries(_dependencyReferences, "projectReference");
```

Comment:
Not sure if calling this "projectReference" is ok., but it allows the user to disambiguate a direct reference from an indirect reference.

Reply:
What do projects have to do with this?

Reply:
Above was written before seeing example.

I don't think we can just invent a name to go there.

And there's not always a project involved, right? What about just single project with `<Reference Include="A.dll">` and A depends on B.dll sitting next to it?

Reply:
`package` is currently used for all package references (direct or indirect)
`project` is currently used for all project references
`reference` is currently used for direct references

I can either have indirect references labeled as `reference` or something else. I had picked `projectReference` since that is what the MSBuild metadata labels it as.

Reply:
I think we're missing cases if we only work with MSBuild projectReference metadata. There can be indirect dependencies that do not come from project references.

Reply:
> I don't think we can just invent a name to go there.

Right, which is why I made sure to call it out. If we can't then I can just relabel it to "reference".

Reply:
> I think we're missing cases if we only work with MSBuild projectReference metadata.

Dependencies of transitive direct references show up the same transitive direct references.

For example:
* `ConsoleApp34` has a `<ProjectReference Include="..\ClassLibrary1\ClassLibrary1.csproj" />`
* `ClassLibrary1` has a `<Reference Include="ClassLibrary2" HintPath="..\ClassLibrary2\bin\Debug\netstandard2.0\ClassLibrary2.dll" />`
* `ClassLibrary2` has a `<ProjectReference Include="..\ClassLibrary3\ClassLibrary3.csproj" />`
* `ClassLibrary3` has no dependencies

`ConsoleApp34` still gets a `ReferenceDependencyPaths` entry for `ClassLibrary3` with a `ReferenceSourceTarget=ProjectReference`

Reply:
OK, I think it should just be "reference". I think things will break if not: 
https://github.com/dotnet/core-setup/blob/61fb41c28a3ec63356df1d03f26444502b7cd19f/src/managed/Microsoft.Extensions.DependencyModel/Resolution/AppBaseCompilationAssemblyResolver.cs#L54

You can still see from the dependency tree when a "project" dragged in a "reference". 

And if there's enough metadata also when a "reference" dragged in another "reference".

Right?

Reply:
Where do we handle this case: What about just single project with <Reference Include="A.dll"> and A depends on B.dll sitting next to it?

What is deps.json for that. It's confusing me to talk about this mix of project references and references straight away.

Reply:
OK, I think I see the distinction in the code now. Tests would be really helpful to understand if all cases are covered.

Reply:
I am still confused, actually. so the question stands.

ConsoleApp has no project references
ConsoleApp has direct dll reference to LibOnDiskA.dll
LibOnDiskA.dll references LibOnDiskB.dll sitting next to it.

Reply:
Do we handle that case?

Reply:
Yes, that is the case I explained here: https://github.com/dotnet/sdk/pull/2090#discussion_r177593683

Reply:
Huh, that has a ProjectReference in it. I am talking about no project references involved at all, but indirect binary reference.

Reply:
Fixed to just be 'reference'

Reply:
Talked to @nguerrera  in person. He was asking about the deps.json for a project that had a direct reference to a dll, which in turn had a dependency (basically what the deps.json for `ClassLibrary1` would look like for https://github.com/dotnet/sdk/pull/2090#discussion_r177593683).

In that case, `ClassLibrary3` shows up in the `ReferenceDependencyPaths` but with `ReferenceSourceTarget=ResolveAssemblyReference` (where-as it shows up as `ReferenceSourceTarget=ProjectReference` in `ConsoleApp34`).

I've updated the logic to not filter the `ReferenceDependencyPaths` nodes by `ReferenceSourceTarget` to account for this.